yt_dlp/extractor/wrestleuniverse.py

```diff
@@ -0,0 +1,231 @@
+                    'start_time': ('start', {int}),
+                    'end_time': ('end', {int}),
+                }),
+            }),
+        }
+
+
+class WrestleUniversePPVIE(WrestleUniverseBaseIE):
+    _VALID_URL = WrestleUniverseBaseIE._VALID_URL_TMPL % 'lives'
+    _TESTS = [{
+        'note': 'HLS AES-128 key obtained via API',
+        'url': 'https://www.wrestle-universe.com/en/lives/buH9ibbfhdJAY4GKZcEuJX',
+        'info_dict': {
+            'id': 'buH9ibbfhdJAY4GKZcEuJX',
+            'ext': 'mp4',
+            'title': '【PPV】Beyond the origins, into the future',
+            'description': 'md5:9a872db68cd09be4a1e35a3ee8b0bdfc',
+            'channel': 'tjpw',
+            'location': '東京・Twin Box AKIHABARA',
+            'duration': 10098,
+            'timestamp': 1675076400,
+            'upload_date': '20230130',
+            'thumbnail': 'https://image.asset.wrestle-universe.com/rJs2m7cBaLXrwCcxMdQGRM/rJs2m7cBaLXrwCcxMdQGRM',
+            'thumbnails': 'count:3',
+            'hls_aes_key': '5633184acd6e43f1f1ac71c6447a4186',
+            'hls_aes_iv': '5bac71beb33197d5600337ce86de7862',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+        },
+    }, {
+        'note': 'unencrypted HLS',
+        'url': 'https://www.wrestle-universe.com/en/lives/wUG8hP5iApC63jbtQzhVVx',
+        'info_dict': {
+            'id': 'wUG8hP5iApC63jbtQzhVVx',
+            'ext': 'mp4',
+            'title': 'GRAND PRINCESS \'22',
+            'description': 'md5:e4f43d0d4262de3952ff34831bc99858',
+            'channel': 'tjpw',
+            'location': '東京・両国国技館',
+            'duration': 18044,
+            'timestamp': 1647665400,
+            'upload_date': '20220319',
+            'thumbnail': 'https://image.asset.wrestle-universe.com/i8jxSTCHPfdAKD4zN41Psx/i8jxSTCHPfdAKD4zN41Psx',
+            'thumbnails': 'count:3',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+        },
+    }]
+
+    _API_PATH = 'events'
+
+    def _real_extract(self, url):
+        lang, video_id = self._match_valid_url(url).group('lang', 'id')
+        metadata = self._download_metadata(url, video_id, lang, 'eventFallbackData')
+
+        info = traverse_obj(metadata, {
+            'title': {lambda v: v.get('displayName') or ''},
+            'description': ('description', {str}),
+            'channel': ('labels', 'group', {str}),
+            'location': ('labels', 'venue', {str}),
+            'timestamp': ('startTime', {int_or_none}),
+            'thumbnails': (('keyVisualUrl', 'alterKeyVisualUrl', 'heroKeyVisualUrl'), {url_or_none}, {'url': None}),
+        })
+
+        ended_time = traverse_obj(metadata, ('endedTime', {int_or_none}))
+        if info.get('timestamp') and ended_time:
+            info['duration'] = ended_time - info['timestamp']
+
+        video_data, decrypt = self._call_encrypted_api(
+            video_id, ':watchArchive', 'watch archive', data={'method': 1})
```

Comment:
Imo, it makes sense for this function to return the decrypted key instead of a function

Reply:
It is cleaner to separate the decryption logic internals from the calling code. The `decrypt` method is also used more than once already. This also encapsulates the respective parts completely, if the en-/decryption changes, only the inner function would need to change. If the api changes, only the outer or call api functions need to change.